src/github.com/getlantern/flashlight/flashlight.go

```diff
@@ -169,6 +177,35 @@ func useAllCores() {
 	runtime.GOMAXPROCS(numcores)
 }
 
+func configureLogging() *rotator.SizeRotator {
+	logdir := appdir.Logs("Lantern")
+	log.Debugf("Placing logs in %v", logdir)
+	if _, err := os.Stat(logdir); err != nil {
+		if os.IsNotExist(err) {
+			// Create log dir
+			if err := os.MkdirAll(logdir, 0755); err != nil {
+				log.Fatalf("Unable to create logdir at %s: %s", logdir, err)
+			}
+		}
+	}
+	file := rotator.NewSizeRotator(filepath.Join(logdir, "lantern.log"))
+	// Set log files to 1 MB
+	file.RotationSize = 1 * 1024 * 1024
+	// Keep up to 20 log files
+	file.MaxRotation = 20
+	errorOut := timestamped(io.MultiWriter(file, os.Stderr))
+	debugOut := timestamped(io.MultiWriter(file, os.Stdout))
+	golog.SetOutputs(errorOut, debugOut)
+	return file
+}
+
+// timestamped adds a timestamp to the beginning of log lines
+func timestamped(orig io.Writer) io.Writer {
+	return wfilter.Lines(orig, func(w io.Writer, line string) (int, error) {
+		return fmt.Fprintf(w, "%s - %v", time.Now().Format(time.RFC3339), line)
```

Comment:
I won't this functionality to golog because I'm following the Go philosophy of keeping libraries small and focused.  golog is just an API for capturing log information at different log levels and writing it to somewhere.  As the `configureLogging` method shows, it is trivial to add behavior like timestamp formatting, log rotation and eventually Loggly by composing writers.